### PR TITLE
Minor devil fixes

### PR DESCRIPTION
**Pull request by lordpidey**
:cl:
add: There is a new potential obligation for devils to have.  Musical duels.  Ask your nearest devil for a musical duel, there's a 14% chance he's obligated to do it.
del: Removed the old offering a drink obligation for devils, it overlapped too much with the food obligation.
tweak: Trying to clone people who've sold their soul now results in FUN.
tweak: Devils can no longer spam ghosts with revival contracts.
/:cl:

This changes a few minor things about devils.  Nothing gamechanging, but minor annoyances/glitches have been fixed.

Replaces OBLIGATION_DRINK with a much more appropriate fiddle reference
Attempting to clone soulsellers no longer works.  Fun things happen if you try.
Fireproof paper no longer becomes unreadable on burning.
Made a generic "Conjure item" spell as a superclass for the summon pitchforks.  These spells summon/unsummon an item directly into your hand.
You can no longer spam ghosts with the infernal revival contract.
Admins can now un-devil a devil, even AFTER they've become a third stage devil.

### CODE REVIEW

**Comment by lordpidey** [comment]
Whoops, I've forgotten to include the new sprites for the fiddle.  I'll add those in the morning, don't merge until then.

File: code/game/machinery/cloning.dm [hunk 1]
```diff
@@ -378,6 +381,15 @@
 	if(!qdeleted(src))
 		go_out()
 
+/obj/machinery/clonepod/proc/horrifyingsound()
+	spawn(0)
```
Comment (optimumtact) [inline code/game/machinery/cloning.dm]: just use add timer(0, "horrifyingsound", src) wherever you're calling this proc

Comment (optimumtact) [inline code/game/machinery/cloning.dm]: hello

**Comment by lordpidey** [comment]
Sorry, I've moved recently and not had a lot of free time.  I'll try to get to fixing the stuff.

**Comment by AnturK** [comment]
@lordpidey Is this ready besides the conflicts ?

**Comment by lordpidey** [comment]
Yeah, it's ready.  Sorry for taking so long with this.

File: code/game/machinery/cloning.dm [hunk 2]
```diff
@@ -120,13 +120,10 @@
 
 /obj/machinery/clonepod/examine(mob/user)
 	..()
-	if (isnull(occupant) || !is_operational())
-		return
-	if ((!isnull(occupant)) && (occupant.stat != DEAD))
+	if(mess)
+		user << "It's filled with blood and vicerea. You swear you can see it moving..."
```
Comment (AnturK) [inline code/game/machinery/cloning.dm]: viscera

File: code/modules/paperwork/paper.dm [hunk 3]
```diff
@@ -339,10 +339,11 @@
 
 	add_fingerprint(user)
 
-/obj/item/weapon/paper/fire_act(exposed_temperature, exposed_volume)
+/obj/item/weapon/paper/fire_act()
```
Comment (AnturK) [inline code/modules/paperwork/paper.dm]: Keep the arguments consistent with the parent proc even if you're not using them please.

File: code/modules/spells/spell_types/devil.dm [hunk 4]
```diff
@@ -1,40 +1,32 @@
...
+/obj/effect/proc_holder/spell/targeted/conjure_item/summon_pitchfork
 	name = "Summon Pitchfork"
 	desc = "A devil's weapon of choice.  Use this to summon/unsummon your pitchfork."
 	invocation_type = "none"
 	include_user = 1
 	range = -1
 	clothes_req = 0
-	var/obj/item/weapon/twohanded/pitchfork/demonic/pitchfork
-	var/pitchfork_type = /obj/item/weapon/twohanded/pitchfork/demonic/
+	item_type = /obj/item/weapon/twohanded/pitchfork/demonic/
```
Comment (AnturK) [inline code/modules/spells/spell_types/devil.dm]: Remove the / at the end of the path please.

**Review by AnturK — CHANGES_REQUESTED** [review]
Minor issues

File: code/game/machinery/cloning.dm [hunk 5]
```diff
@@ -121,7 +121,7 @@
 /obj/machinery/clonepod/examine(mob/user)
 	..()
 	if(mess)
-		user << "It's filled with blood and vicerea. You swear you can see it moving..."
+		user << "It's filled with blood and viscrea. You swear you can see it moving..."
```
Comment (optimumtact) [inline code/game/machinery/cloning.dm]: viscera

File: code/game/machinery/cloning.dm [hunk 6]
```diff
@@ -120,13 +120,10 @@
 
 /obj/machinery/clonepod/examine(mob/user)
 	..()
-	if (isnull(occupant) || !is_operational())
-		return
-	if ((!isnull(occupant)) && (occupant.stat != DEAD))
+	if(mess)
+		user << "It's filled with blood and viscrea. You swear you can see it moving..."
```
Comment (optimumtact) [inline code/game/machinery/cloning.dm]: viscera

File: code/game/machinery/cloning.dm (hunk 5 above)
Comment (PKPenguin321) [inline code/game/machinery/cloning.dm]: viscereal

**Comment by KingPhilipIII** [comment]
Definitely approve with the pitchfork. I'd always summon it, then lose it for one reason or another, and it's gone.

**Review by AnturK — APPROVED** [review]

**Comment by lordpidey** [comment]
That's how it worked before?  Press button -> It appears.  Press it again, it dissappears, press it again, it appears.

**Comment by KingPhilipIII** [comment]
No. before if you lost it it never despawned until you changed form. Id lose my pitchfork as an arch devil and it would just be gone.

**Comment by KingPhilipIII** [comment]
Another thing Pidey. If you're tweaking your antag, I personally love playing as a devil. It's loads of fun, and I've had some meaningful role play (I've convinced the HoS to sell his soul. No criminals can hide from X-ray cops.) but when the devil becomes a true or arch devil, they need to be able to control the spawn of imps or banish their imps with a command. Every time I ascend I have a horde of imps slaughtering everyone while asking for promotions even when I order them not to attack anyone who sold their souls, or on the occasion I promised not to harm the crew everyone. It's somewhat ridiculous.

**Comment by lordpidey** [comment]
`/obj/effect/proc_holder/spell/targeted/summon_pitchfork/cast(list/targets, mob/user = usr)`
 	`if (pitchfork)`
 		`qdel(pitchfork)`
 	`else`
 		`for(var/mob/living/carbon/C in targets)`
 			`if(C.drop_item())`
 				`pitchfork = new pitchfork_type`
 				`C.put_in_hands(pitchfork)`

That's the old code.  Unless something is going wrong with the qdel, it SHOULD work.  If the pitchfork exists, destroy it, if it does NOT exist, put it in your hands.  THe only reason it should have failed before is if somehow both of your hands can't drop something.  I'll look into it.  But what you're describing isn't intended behavior.

**Comment by KingPhilipIII** [comment]
I rarely summoned before becoming a true devil, at which point the crew becomes aggressive and I need to defend myself. Might have something to do with it.

**Comment by KingPhilipIII** [comment]
I'd just lose it for some reason, resort to fireballing my attackers, and someone would charge me with my own pitchfork at which point I kill them and take it back or I click the button and it summons to my hand. I have no idea how it worked.

**Comment by lordpidey** [comment]
I looked at the old code some more, and the new code is pretty similar, but it MIGHT fix the bug you are talking about.

I wasn't able to reproduce the bug on the old, or the new code, but I think adding the `item = null` line might fix it.  Time will tell.

**Comment by KingPhilipIII** [comment]
Would the fact I was never actually a natural devil matter? Code makes zero sense to me, so it's probably irrelevant, Admins always made me one though.

**Comment by lordpidey** [comment]
No.

Now that I think about it, the bug will only occur when the server is stressed, and the qdel system is delayed, which would explain why local tests didn't find it.

**Comment by KingPhilipIII** [comment]
So. With my limited understanding of what you just said. The server relegated destruction of the pitchfork to a time where it was less stressed, and it basically ceased to exist. And then it searched for the wrong criteria onto whether or not it had destroyed the pitchfork?

**Comment by lordpidey** [comment]
Yes, that's exactly it.

**Comment by KingPhilipIII** [comment]
When I think about it. Four of the five times I ascended the station was on fire or half blown up or the engine had escaped. So that's a pretty logical conclusion.

**Pull request merged**